Integrations/MISP_V2/MISP_V2.py

```diff
@@ -128,6 +125,71 @@ def warn(*args):
 ''' HELPER FUNCTIONS '''
 
 
+def filter_obj(obj: dict, is_att: bool = True) -> dict:
+    """ Filter Event, Attribute
+
+    Args:
+        obj(dict): dictionary MISP event or attribute.
+        is_att: True if dict represent Attribute otherwise Event (False)
+
+    Returns:
+        dict: Filtered dict by configuration.
+    """
+    metadata_state = demisto.params().get('metadata')
+    related_events_state = demisto.params().get('related_events')
+    selected_keys: List[str] = [item.lower() for item in demisto.params().get('context_select')]
+    ignored_keys_metadata: List[str] = ["Galaxy", "Tag"]
+    ignored_keys_related_events: List[str] = ["RelatedEvent"]
+
+    for dict_key in list(obj.keys()):
+        if dict_key in ignored_keys_metadata:
+            if not metadata_state:
+                obj[dict_key] = []
+            else:
+                continue
+        elif dict_key in ignored_keys_related_events:
+            if not related_events_state:
+                obj[dict_key] = []
+            else:
+                continue
+        elif is_att and selected_keys and dict_key not in selected_keys:
+            obj.pop(dict_key)
+
+    return obj
+
+
+def filter_misp_response(misp_response: List[dict]) -> List[dict]:
+    """ Filter MISP response -
+        Remove from Attributes and Objects the following keys if exists:
+         1. MetaData keys - Galaxy, Tags (The keys both determine classification by publisher)
+         2. Selected keys - Removing all not selected keys in response.
+         3. Related events - Remove related events if not chosen.
+
+    Args:
+        misp_response(list): valid response of MISP client using PyMisp
+
+    Returns:
+        list: Filtered response
+    """
+    metadata_state = demisto.params().get('metadata')
+    related_events_state = demisto.params().get('related_events')
+    selected_keys: List[str] = [item.lower() for item in demisto.params().get('context_select')]
+
+    if not related_events_state or not metadata_state or selected_keys:
+        for i in range(len(misp_response)):
```

Comment:
this looks like C...
```suggestion
        for res in misp_response:
```
replace all `misp_response[i]` with `res`